pos_deposit/models/product_template.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2024 Hunki Enterprises BV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    is_deposit = fields.Boolean(
+        help="Check this if this product is a container for which you charge deposit",
+    )
+    deposit_product_id = fields.Many2one(
+        "product.product",
+        "Deposit",
+        domain=[("is_deposit", "!=", False)],
+        help="If this product is packaged in a container for which you charge deposit, "
+        "add a product here that stands for the deposit",
+        compute="_compute_deposit_product_id",
+        inverse="_inverse_deposit_product_id",
+        search="_search_deposit_product_id",
+    )
+
+    @api.depends("product_variant_ids.deposit_product_id")
+    def _compute_deposit_product_id(self):
+        for this in self:
+            this.deposit_product_id = this.product_variant_ids.deposit_product_id
+
+    def _inverse_deposit_product_id(self):
+        for this in self:
+            this.product_variant_ids.write(
+                {
+                    "deposit_product_id": this.deposit_product_id,
+                }
+            )
+
+    def _search_deposit_product_id(self, operator, value):
+        return [("product_variant_ids.deposit_product_id", operator, value)]
+
```

Comment:
```suggestion

    def _get_related_fields_variant_template(self):
        res = super()._get_related_fields_variant_template()
        res.append("deposit_product_id")
        return res

```

fields on product and template should be declared in this function. Otherwise, there is some trouble.